lib/rubocop/cop/performance/string_chars.rb

```diff
@@ -0,0 +1,110 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Performance
+      # This cop identifies places where `String#chars.<method_name>(args)`
+      # can be replaced by `String#<method_name>(args)`.
+      #
+      # @example
+      #   # bad
+      #   str.chars[0..2]
+      #   str.chars.slice(0..2)
+      #   str.chars.first
+      #   str.chars.last
+      #   str.chars.length
+      #   str.chars.size
+      #
+      #   # good
+      #   str[0..2]
+      #   str.slice(0..2)
+      #   str[0]
+      #   str[-1]
+      #   str.length
```

Comment:
I think that only compatible methods should be targeted.

```ruby
str = 'abcdefg'

puts '### incompatible'
puts '# str.chars.slice(0..2)'
p str.chars.slice(0..2)
puts '# str.slice(0..2)'
p str.slice(0..2)
puts

puts '### incompatible'
puts '# str.chars[0..2]'
p str.chars[0..2]
puts '# str[0..2]'
p str[0..2]
puts

puts '### compatible'
puts '# str.chars.first'
p str.chars.first
puts '# str[0]'
p str[0]
puts

puts '### compatible'
puts '# str.chars.last'
p str.chars.last
puts '# str[-1]'
p str[-1]
puts

puts '### compatible'
puts '# str.chars.length'
puts str.chars.length
puts '# str.length'
puts str.length
puts

puts '### compatible'
puts '# str.chars.size'
puts str.chars.size
puts '# str.size'
puts str.size
```

```console
% ruby example.rb
### incompatible
# str.chars.slice(0..2)
["a", "b", "c"]
# str.slice(0..2)
"abc"

### incompatible
# str.chars[0..2]
["a", "b", "c"]
# str[0..2]
"abc"

### compatible
# str.chars.first
"a"
# str[0]
"a"

### compatible
# str.chars.last
"g"
# str[-1]
"g"

### compatible
# str.chars.length
7
# str.length
7

### compatible
# str.chars.size
7
# str.size
7
```

Reply:
> I think that only compatible methods should be targeted.

Sure! I was dumb missing that `slice`ing and `[...]`ing an array actually returns an array 😄 
Thanks.